src/uct/base/uct_iface.h

```diff
@@ -353,25 +354,72 @@ typedef void (*uct_iface_mpool_init_obj_cb_t)(uct_iface_h iface, void *obj,
 
 
 /**
- * Base structure for private data held inside a pending request. Contains
- * a queue element so we can put this on a queue.
+ * Base structure for private data held inside a pending request for TLs
+ * which use ucs_arbiter_t to progress pending requests.
  */
 typedef struct {
-    ucs_queue_elem_t  queue;
-} uct_pending_req_priv_t;
+    ucs_arbiter_elem_t  arb_elem;
+} uct_pending_req_priv_arb_base_t;
```

Comment:
uct_pending_req_priv_arb_t
i would assume some transports could use it directly, not just extend it